histomicstk/features/compute_global_graph_features.py

```diff
@@ -0,0 +1,168 @@
+from __future__ import division
+
+from collections import namedtuple
+from itertools import combinations
+
+import numpy as np
+from numpy import linalg
+from scipy.spatial import cKDTree as KDTree, Voronoi
+from scipy import sparse
+from scipy.sparse.csgraph import minimum_spanning_tree
+
+PopStats = namedtuple('PopStats', ['mean', 'stddev', 'minmaxr', 'disorder'])
+PolyProps = namedtuple('PolyProps', ['area', 'peri', 'max_dist'])
+TriProps = namedtuple('TriProps', ['sides', 'area'])
+DensityProps = namedtuple('DensityProps', ['neighbors_in_distance',
+                                           'distance_for_neighbors'])
+Props = namedtuple('Props', ['voronoi', 'delaunay', 'mst_branches', 'density'])
+
+
+def compute_global_graph_features(centroids, scale=10., neighbor_counts=(3, 5, 7)):
+    """Compute global (i.e., not per-pixel) graph-based features of the
+    nuclei with the given centroids.
+
+    Parameters
+    ----------
+    centroids : array_like
+        Nx2 numpy array of nuclear centroids
+    scale : float
+        Multiplier for distances.  Used for determining the radii to
+        count neighbors in.
```

Comment:
Use an array of radii instead rather internally generating it with `scale * np.arange(6)`. You can use the settings in the paper as the default value.